src/components/Main/MainView/MessageInput/use/wordSuggester.ts

```diff
@@ -0,0 +1,101 @@
+import store from '@/store'
+import createTree from '@/lib/trieTree'
+import { animeEffectSet, sizeEffectSet } from '@traptitech/traq-markdown-it'
+import { ComputedRef, WritableComputedRef, Ref, ref } from 'vue'
+import textFieldMirror from './textFieldMirror'
+
+export type Target = {
+  word: string
+  begin: number
+  end: number
+  divided: boolean
+}
+
+const getCurrentWord = (elm: HTMLTextAreaElement, text: string): Target => {
+  text = text.replaceAll('　', ' ')
+  const startIndex = elm.selectionStart
+  const prevAtMarkIndex = text.lastIndexOf('@', startIndex - 1)
+  const prevColonIndex = text.lastIndexOf(':', startIndex - 1)
+  const prevPeriodIndex = text.lastIndexOf('.', startIndex - 1)
+  const nearest = Math.max(prevAtMarkIndex, prevColonIndex, prevPeriodIndex)
+  const begin = nearest < 0 ? 0 : nearest
+  const end = elm.selectionEnd
+  const word = text.substring(begin, end)
+  const prevSpaceIndex = text.lastIndexOf(' ', startIndex - 1)
+  const divided = prevSpaceIndex > nearest
+  return { word, begin, end, divided }
+}
+
+const useWordSuggester = (
+  textareaRef: ComputedRef<HTMLTextAreaElement | undefined>,
+  value: WritableComputedRef<string>
+) => {
+  const hideSuggester = ref(true)
+  const interactingWithList = ref(false)
+  const position = ref({ top: 0, left: 0 })
+  const target = ref({
+    word: '',
+    begin: 0,
+    end: 0,
+    divided: false
+  }) as Ref<Target>
+  const suggesteCandidates = ref([] as string[])
```

Comment:
```suggestion
  const target = ref<Target>({
    word: '',
    begin: 0,
    end: 0,
    divided: false
  })
  const suggesteCandidates = ref<string[]>([])
```
`as`は使わなくて済むならそっちのが安全なので